src/v/pandaproxy/server.cc

```diff
@@ -249,4 +253,69 @@ ss::future<> server::stop() {
       .finally([this]() mutable { return _server.stop(); });
 }
 
+server::probe::probe(context_t& ctx, const ss::sstring& group_name)
+  : _ctx(ctx)
+  , _group_name(group_name)
+  , _metrics()
+  , _public_metrics() {
+    setup_metrics();
+}
+
+void server::probe::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    auto setup_common = [this]<typename MetricDef>() {
+        const auto used_units = [](const size_t current, const size_t max) {
+            return max - current;
```

Comment:
I think it's useful to return this normalized [0,1], since we don't have a metric for the limit.

Reply:
The question is how would you set an alert? I'm happy to leave it at an absolute value, but also include a metric for the limit.

Reply:
> The question is how would you set an alert?

Valid point.

> but also include a metric for the limit.

i think having a normalized value and the limits is the best of both worlds. Easy monitoring on how full your system is and my previous comment goes away.

Reply:
i ended up going with just the normalized metrics. It sounds too much to have the maximum values as well, just for an edge case.